clients/stellar-relay-lib/src/connection/connector/connector.rs

```diff
@@ -32,11 +36,8 @@ pub struct Connector {
 	handshake_state: HandshakeState,
 	flow_controller: FlowController,
 
-	/// a channel for writing xdr messages to stream.
-	actions_sender: mpsc::Sender<ConnectorActions>,
-
-	/// a channel for communicating back to the caller
-	relay_message_sender: mpsc::Sender<StellarRelayMessage>,
+	/// for writing xdr messages to stream.
+	pub(crate) write_stream_overlay: OwnedWriteHalf,
```

Comment:
instead of interpreting the message because of its wrapper (`ConnectorActions`, `StellarRelayMesssage`), the `Connector` will receive a `StellarMessage` from the user and send that immediately to the Node.